### DISCUSSION TITLE
Prerequsite Required Dialog

### DISCUSSION THREAD

**Discussion by chrpai** [wixtoolset/issues#7380] [Questions] [answered]
I'm getting my feet wet with WixInternalUIBootstrapperApplication and loving it.  Incredibly useful to me despite.  But I'm wondering, how customizable is this dialog?

![image](https://user-images.githubusercontent.com/43178814/230694414-4fcbe55c-905b-4e89-85d5-2f5d62b8a615.png)

Is it possible to get it to display a list of the prereqs that are planned to be installed?  Something akin to:

![image](https://user-images.githubusercontent.com/43178814/230694503-f81d32cf-520d-4740-8731-cdc414663451.png)

**Comment by rseanhall** [comment]
The short answer is that it's the same way that you customize `WixStandardBootstrapperApplication` - the XML theme file for declarative customization and the optional BAFunction DLL for running code. The standard theme file is at https://github.com/wixtoolset/wix4/blob/6f3997198d894d60afce07bd6a68bcad5fd20ba0/src/ext/Bal/wixstdba/Resources/iuipreq.thm.

You're not going to be able to dynamically show the list of prereqs or their status without writing code.

**Comment by chrpai** [comment]
How much work are we talking?  I'm willing to sponsor/hire someone to get this into WiX or just into my own custom project templates.  It's important to me to tell the user what prereqs are to be installed prior to them accepting.

**Reply by rseanhall** [reply, marked as answer]
It wouldn't be too hard. But I expected people that are using this BA to just hardcode it in the theme if they wanted to show that kind of information. I guess if all the prereqs are ExePackages then you could probably use the detect conditions in creative ways to make it a little less hardcoded.
